### ISSUE TITLE
ExoMol Support

### ISSUE THREAD

**Issue by erwanp** [radis/radis#319]
### 🎯 Describe what you'd like

ExoMol support. Asked for in #34 

### Implementation (see https://github.com/radis/radis/pull/320)

```python
from radis import SpectrumFactory

sf = SpectrumFactory(
        4310,
        4320,
        molecule="H2O",
        isotope="1",
        verbose=3,
    )
sf.fetch_databank("exomol")
s = sf.eq_spectrum(500)
s.plot()
```

![image](https://user-images.githubusercontent.com/16088743/126080498-5f437d32-c274-47bf-a17a-995ad7f3d4a7.png)

### 💡 Possible implementations

Borrowing and adapting some codes from the [ExoJax](https://github.com/HajimeKawahara/exojax) project from @HajimeKawahara (which you should also have a look at !) Hopefully we can end up maintaining a unified code base for local database formats (currently : ExoJax uses Feather format; which is extremelly efficient but cannot be masked/filtered as easily as HDF5)

---
<sub>*Tip : you can vote for [New Features](https://github.com/radis/radis/issues/79) on https://feathub.com/radis/radis*</sub>

**Comment by minouHub** [comment]
@erwanp <3

I tried your example. Long download as expected but works amazingly well. I wanted to try with H2CO because I'm interested in that molecule at the moment. I just added it and it seems to work. Do you have an idea of what to do to automatize it for all molecules or should add them as it comes? 

Note: there is an error later in the `pickup_gE` fr H2CO. I'll take a look at it.

```python
KNOWN_EXOMOL_ISOTOPES_NAMES = {
    "H2O": {1: "1H2-16O"},
    "NH3": {1: "14N-1H3"},
    "H2S": {1: "1H2-32S"},
    "FeH": {1: "56Fe-1H"},
    "H2CO": {1: "1H2-12C-16O"},
}
"""molecule: isotope: full name"""
KNOWN_EXOMOL_DATABASE_NAMES = {
    "H2O": {"1H2-16O": ["POKAZATEL"]},
    "NH3": {"14N-1H3": ["CoYuTe"]},
    "H2S": {"1H2-32S": ["AYT2"]},
    "FeH": {"56Fe-1H": ["MoLLIST"]},
    "H2CO": {"1H2-12C-16O": ["AYTY"]},
}
```

**Comment by erwanp** [comment]
Great !! 

It can definitly be automatized, the hardcoded dictionaries above were only for the tests. The only things missing to automatize are : 
- the correspondance between HITRAN's isotopes ranking for Earth abundances (1, 2, etc..)  and the full name of the molecule in ExoMol (ex : H2O,2 = 1H2-18O ). This should be quite easy to write from [molparams](https://github.com/radis/radis/blob/develop/radis/db/molparam.txt) for all HITRAN molecules ; I do not know if there are molecules in ExoMol not defined in HITRAN.
- the dynamic generation of the list of all databases available for a given isotope (ex : HotWat78 for the above) 

I'll look it up further.

**Comment by HajimeKawahara** [comment]
@erwanp Do you mean [this one](http://secondearths.sakura.ne.jp/exojax/exojax/exojax.utils.html#module-exojax.utils.molname)?

For the exomol support, it might be good to check the recent update in the "develop" branch of exojax: see [here](https://github.com/HajimeKawahara/exojax/pull/99).

**Comment by erwanp** [comment]
Indeed, [this seems related](https://github.com/HajimeKawahara/exojax/pull/99#issuecomment-881311860) to the `pickup_gE` problem encountered by @minouHub 

@HajimeKawahara for the full name I found e2s, very useful. I'm also looking forward to the list of all databases referenced on ExoMol for a given isotope. For instance `1H2-18O 👉  HotWat78 ` , `1H2-16O 👉  BT2, VT, POKAZATEL, etc. `

**Comment by erwanp** [comment]
@minouHub RADIS now automatically finds the available ExoMol databases (and recommended one) https://github.com/radis/radis/pull/320/commits/3ab6f0b1d98e68a3bfeb871bc92f6a73376cc387#diff-0ccf9b5bb7915e8940d04891456118106d4cc97e3052e8d77fc2b7b85673ba6bR66.  @HajimeKawahara if you want to borrow for ExoJax !

**Comment by HajimeKawahara** [comment]
@erwanp it's cool! Thanks.
